pkg/storaged/part-tab.jsx

```diff
@@ -51,6 +76,22 @@ export class PartitionTab extends React.Component {
                     <DescriptionListDescription>{block_part.Type}</DescriptionListDescription>
                 </DescriptionListGroup>
             </DescriptionList>
-        );
-    }
-}
+            { unused_space &&
+            <>
+                <br />
+                <Alert variant="warning"
+                         isInline
+                         title={_("This partition is not completely used by its content.")}>
+                    {cockpit.format(_("Partition size is $0. Content size is $1."),
+                                    utils.fmt_size(unused_space_warning.volume_size),
+                                    utils.fmt_size(unused_space_warning.content_size))}
+                    <div className='storage_alert_action_buttons'>
+                        <StorageButton excuse={shrink_excuse} onClick={shrink}>{_("Shrink partition")}</StorageButton>
+                        <StorageButton excuse={grow_excuse} onClick={grow}>{_("Grow content")}</StorageButton>
+                    </div>
+                </Alert>
```

Comment:
These 12 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19385-20230925-111150-44391ef2-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)